lib/services/downloads_service.dart

```diff
@@ -681,15 +683,56 @@ class DownloadsService {
     // Step 4 - Make sure there are no unanchored nodes in metadata.
     _downloadsLogger.info("Starting downloads repair step 4");
     downloadCounts[repairStepTrackingName] = 4;
+    final Map<int, LyricDto?> idsWithLyrics = HashMap();
+    var allItems = _isar.downloadItems.where().findAllSync();
+    for (var item in allItems) {
+      if (item.baseItem?.mediaStreams?.any((stream) => stream.type == "Lyric") ?? false) {
+        // check if lyrics are already downloaded
+        if (_isar.downloadedLyrics.where().isarIdEqualTo(item.isarId).countSync() > 0) {
+          continue;
+        }
+        idsWithLyrics[item.isarId] = null;
+      }
+    }
+    //TODO move this into a separate function in the background service
+    final JellyfinApiHelper _jellyfinApiData = GetIt.instance<JellyfinApiHelper>();
+    for (var id in idsWithLyrics.keys) {
+      var canonItem = _isar.downloadItems.getSync(id);
+      if (canonItem?.baseItem != null) {
+        LyricDto? lyrics;
+        try {
+          lyrics = await _jellyfinApiData.getLyrics(itemId: canonItem!.baseItem!.id);
+          _downloadsLogger.finer("Fetched lyrics for ${canonItem!.baseItem!.name}");
+          idsWithLyrics[id] = lyrics;
+        } catch (e) {
+          _downloadsLogger.warning("Failed to fetch lyrics for ${canonItem!.baseItem!.name}.");
+          rethrow;
```

Comment:
Errors in the downloads repair should probably be caught, to make sure we get through all the steps.